### PR TITLE
[categories] Added "Hot Spring".

### PR DESCRIPTION
**Pull request by vng**

### CODE REVIEW

**Review by biodranik — APPROVED** [review]
LGTM

@organicmaps/translations can you please check and suggest good synonyms that users may type in the search to find natural spring?

File: data/categories.txt [hunk 1]
```diff
@@ -9077,6 +9077,46 @@ sw:4Maji ya kunywa
...
+it:Primavera calda
+ja:温泉
+ko:온천
+mr:गरम पाण्याचा झरा
+nb:Varm kilde
+nl:Warme lente|Heetwaterbron
+pl:Gorące źródło
+pt:Primavera quente
+pt-BR:Primavera quente
+ro:Izvorul termal
```
Comment (vgrigoriu) [inline data/categories.txt:9108]: `Izvor termal` (there's no need for the definite article).

File: data/categories.txt [hunk 2]
```diff
@@ -9077,6 +9077,46 @@ sw:4Maji ya kunywa
...
+en:3Hot Spring
+ar:ربيع حار
+be:4Гарачая крыніца|4Геатэрмальная крыніца
+bg:Гореща пролет
+ca:Primavera càlida
+cs:Teplé jaro|Termální pramen
+da:Varm kilde
+de:Heiße Quelle|Thermalquelle
+el:Ιαματικό νερό
+es:Primavera calurosa
```
Comment (fitojb) [inline data/categories.txt:9090]: It should be "Aguas termales".

File: data/categories.txt [hunk 3]
```diff
@@ -9077,6 +9077,46 @@ sw:4Maji ya kunywa
...
+et:Kuumaveeallikas
+eu:Udaberri beroa
+fa:چشمه آب گرم
+fi:Kuuma lähde
+fr:Source chaude
+he:מעיין חם
+hu:Forró tavasz
```
Comment (d4f5409d) [inline data/categories.txt:9097]:
```suggestion
hu:Melegvizű forrás
```

**Review by d4f5409d — CHANGES_REQUESTED** [review]
😂 this is a funny one! "Tavasz" means the time of the year/season  'spring' in Hungarian, not the water spring (forrás)

**Review by biodranik — CHANGES_REQUESTED** [review]
If the initial Deepl translation gives incorrect result in some language, then use an explicit translation in another language that can't be interpreted differently to fix the issue:
```bash
tools/python/translate.py "ru:Горячий источник"
```


Here is the list of correct translations that should be used/reviewed:

```
en:Hot Spring
ar:ربيع حار
be:Гарачая крыніца
bg:Горещ извор
ca:Primavera càlida
cs:Vřídelní pramen
da:Varmt forår
de:Heiße Quelle
el:Ζεστή άνοιξη
es:Aguas termales
es-MX:Primavera calurosa
et:Kuum allikas
eu:Udaberri beroa
fa:چشمه آب گرم
fi:Kuuma lähde
fr:Source d'eau chaude
he:מעיין חם
hu:Forró forrás
id:Pemandian Air Panas
it:Primavera calda
ja:温泉
ko:온천
mr:गरम पाण्याचा झरा
nb:Varm kilde
nl:Hete Bron
pl:Gorące źródło
pt:Fonte quente
pt-BR:Fontes termais
ro:Izvorul fierbinte
ru:Горячий источник
sk:Horúci prameň
sv:Varm källa
sw:Majira ya joto
th:น้ำพุร้อน
tr:Kaplıca
uk:Гаряче джерело
vi:Suối nóng
zh-Hans:温泉
zh-Hant:溫泉
============ strings.txt format ============
    en = Hot Spring
    af = Warmbron
    ar = ربيع حار
    be = Гарачая крыніца
    bg = Горещ извор
    ca = Primavera càlida
    cs = Vřídelní pramen
    da = Varmt forår
    de = Heiße Quelle
    el = Ζεστή άνοιξη
    es = Aguas termales
    es-MX = Primavera calurosa
    et = Kuum allikas
    eu = Udaberri beroa
    fa = چشمه آب گرم
    fi = Kuuma lähde
    fr = Source d'eau chaude
    he = מעיין חם
    hi = गर्म झरना
    hu = Forró forrás
    id = Pemandian Air Panas
    it = Primavera calda
    ja = 温泉
    ko = 온천
    lt = Karštasis šaltinis
    mr = गरम पाण्याचा झरा
    nb = Varm kilde
    nl = Hete Bron
    pl = Gorące źródło
    pt = Fonte quente
    pt-BR = Fontes termais
    ro = Izvorul fierbinte
    ru = Горячий источник
    sk = Horúci prameň
    sv = Varm källa
    sw = Majira ya joto
    th = น้ำพุร้อน
    tr = Kaplıca
    uk = Гаряче джерело
    vi = Suối nóng
    zh-Hans = 温泉
    zh-Hant = 溫泉
```

File: data/categories.txt [hunk 4]
```diff
@@ -9077,6 +9077,46 @@ sw:4Maji ya kunywa
...
+natural-hot_spring|@water
+en:3Hot Spring
+ar:ربيع حار
+be:4Гарачая крыніца|4Геатэрмальная крыніца
+bg:Гореща пролет
+ca:Primavera càlida
+cs:Teplé jaro|Termální pramen
+da:Varm kilde
+de:Heiße Quelle|Thermalquelle
+el:Ιαματικό νερό
```
Comment (cryoranger) [inline data/categories.txt:9089]: Ιαματικές πηγές

**Review by Wivik — APPROVED** [review]
OK for French

File: data/categories.txt [hunk 5]
```diff
@@ -9077,6 +9077,46 @@ sw:4Maji ya kunywa
...
+pt:Primavera quente
+pt-BR:Primavera quente
+ro:Izvorul termal
+ru:4Горячий источник|4Геотермальный источник
+sk:Horúci prameň
+sv:Varm källa
+sw:Majira ya joto
+th:น้ำพุร้อน
+tr:Kaplıca
+uk:4Гаряче джерело|4Геотермальне джерело
```
Comment (strump) [inline data/categories.txt:9115]:
```suggestion
uk:4Гаряче джерело|4Геотермальне джерело|4Термальні води
```

**Review by strump — COMMENTED** [review]
It could be also named ["Термальні води"](https://uk.wikipedia.org/wiki/Термальні_води)

**Comment by vng** [comment]
Please, review one more time and one more category. Thanks!

File: data/categories.txt [hunk 6]
```diff
@@ -14789,6 +14828,46 @@ tr:İskele|Marina|Rıhtım
...
+id:Ski|Seluncur salju|Lari ski
+it:Sci|Slittino|Pista da sci
+ja:スキー|ソリ|スキー滑走
+ko:스키|썰매|스키 달리기
+mr:स्कीइंग|स्लेडिंग|स्की रन
+nb:Skikjøring|Kjelkekjøring|Skiløype
+nl:Skiën|Sleeën|Skibaan
+pl:Narciarstwo|Saneczkarstwo|Stok narciarski
+pt:Esquiar|Sledding|Corrida de esqui
+ro:Schi|Săniuș|Piste de schi
```
Comment (vgrigoriu) [inline data/categories.txt:14859]:
```suggestion
ro:Schi|Săniuș|Pârtie de schi
```

File: data/categories.txt [hunk 7]
```diff
@@ -9077,6 +9077,45 @@ sw:4Maji ya kunywa
...
+natural-hot_spring|@water
+en:3Hot Spring
+ar:ربيع حار
+be:4Гарачая крыніца|4Геатэрмальная крыніца
+bg:Горещ извор
+ca:Font termal|Font calenta|Deu termal
+cs:Termální pramen
+da:Varm kilde
+de:Heiße Quelle|Thermalquelle
+el:Ιαματικές πηγές
```
Comment (cryoranger) [inline data/categories.txt:9089]: after some more research I think it is better to have it like this:
```suggestion
el:Θερμή πηγή
```

Comment (vng) [reply data/categories.txt:9089]: We can keep both variants for search (categories), but keep only one for description (type_strings.txt)

File: data/categories.txt [hunk 8]
```diff
@@ -14789,6 +14828,46 @@ tr:İskele|Marina|Rıhtım
...
+fr:Ski|Luges|Piste de ski
+he:סקי|מזחלות|ריצת סקי
+hu:Síelés|Szánkózás|Sípálya
+id:Ski|Seluncur salju|Lari ski
+it:Sci|Slittino|Pista da sci
+ja:スキー|ソリ|スキー滑走
+ko:스키|썰매|스키 달리기
+mr:स्कीइंग|स्लेडिंग|स्की रन
+nb:Skikjøring|Kjelkekjøring|Skiløype
+nl:Skiën|Sleeën|Skibaan
```
Comment (jimcarst) [inline data/categories.txt:14856]:
```suggestion
nl:Skiën|Sleeën|Skipiste
```

File: data/categories.txt [hunk 9]
```diff
@@ -14789,6 +14828,46 @@ tr:İskele|Marina|Rıhtım
 uk:5Пристань|3Марина
 zh-Hans:游艇码头
 
+piste:type
+en:3Skiing|Sledding|Ski run
```
Comment (LaoshuBaby) [inline data/categories.txt:14832]: "3Skiing"?

**Review by LaoshuBaby — APPROVED** [review]

File: data/categories.txt (hunk 9 above)
Comment (vng) [reply data/categories.txt:14832]: 3 is a prefix chars number. After entering "Ski" the app will suggest "Skiing" category.

File: data/categories.txt (hunk 7 above)
Comment (cryoranger) [reply data/categories.txt:9089]: then I would suggest to have `Θερμή πηγή` in `type_strings.txt` and `Θερμή πηγή|Ιαματική πηγή` in the search categories.

**Review by d4f5409d — APPROVED** [review]
🇭🇺 Lgtm

File: data/categories.txt [hunk 10]
```diff
@@ -14789,6 +14828,46 @@ tr:İskele|Marina|Rıhtım
...
+en:3Skiing|Ski run
+ar:التزلج|مسار التزلج
+be:Лыжы|Лыжная траса
+bg:Ски|Ски писта
+ca:Esquí|Pista d'esquí
+cs:Lyžování|Lyžařská sjezdovka
+da:Skiløb|Skiløjpe
+de:Skifahren|Skipiste
+el:Σκι|Πίστα σκι
+es:Esquí|Esquí de pista
```
Comment (fitojb) [inline data/categories.txt:14841]: It should read "Pista de esquí" here too :)

File: search/search_integration_tests/smoke_test.cpp [hunk 11]
```diff
@@ -223,6 +223,8 @@ UNIT_CLASS_TEST(SmokeTest, CategoriesTest)
       {"highway", "trunk_link"},
       {"highway", "unclassified"},
       {"historic", "citywalls"},
+      {"piste:type", "downhill"},
+      {"piste:type", "nordic"},
```
Comment (biodranik) [inline search/search_integration_tests/smoke_test.cpp:227]: snow_park, playground, skitour, even better - add everything. Do we differentiate them in types? This is important.

File: data/categories.txt [hunk 12]
```diff
@@ -14789,6 +14828,46 @@ tr:İskele|Marina|Rıhtım
 uk:5Пристань|3Марина
 zh-Hans:游艇码头
 
+piste:type-downhill|piste:type-nordic
```
Comment (biodranik) [inline data/categories.txt:14831]: Let's make a UI category for them like @food, there will be many other types.

File: search/search_integration_tests/smoke_test.cpp (hunk 11 above)
Comment (vng) [reply search/search_integration_tests/smoke_test.cpp:227]: We don't have minor piste types.

File: data/categories.txt (hunk 12 above)
Comment (vng) [reply data/categories.txt:14831]: Maybe next time for the UI, not in this PR.

File: data/categories.txt [hunk 13]
```diff
@@ -9077,6 +9077,45 @@ sw:4Maji ya kunywa
 fa:اب اشامیدنی
 mr:पिण्याचे पाणी
 
+natural-hot_spring|@water
+en:3Hot Spring
+ar:ربيع حار
+be:4Гарачая крыніца|4Геатэрмальная крыніца
```
Comment (biodranik) [inline data/categories.txt:9083]: Will it be found if a user types крынiца or spring? IMO it should be found.

File: search/search_integration_tests/smoke_test.cpp (hunk 11 above)
Comment (biodranik) [reply search/search_integration_tests/smoke_test.cpp:227]: @vng those are not minor, they are different ones. There are fewer of them, but they are still very interesting for many skiers. Is it hard to add them? How to do that?

**Pull request merged**